File: fizzbuzz.hs
-- Language:       Haskell
-- Web site:       http://www.haskell.org/
-- Last tested on: Linux Mint 17.3
-- Requires:       apt-get install ghc

line n
    | n `mod` 15 == 0 = "FizzBuzz"
    | n `mod`  3 == 0 = "Fizz"
    | n `mod`  5 == 0 = "Buzz"
    | otherwise       = show n

main = mapM (putStrLn . line)[1..100]

File: fizzbuzz.pas
program FizzBuzz;
var
    I: Integer;
begin
    for I := 1 to 100 do
    begin
        if I mod 15 = 0 then
            writeln('FizzBuzz')
        else if I mod 3 = 0 then
            writeln('Fizz')
        else if I mod 5 = 0 then
            writeln('Buzz')
        else
            writeln(I);
    end;
end.